_posts/2018/2018-04-01-stop-robots-crawlers-triggering-errors-rails.md

```diff
@@ -0,0 +1,98 @@
+---
+title: 'Stop robots and crawlers causing errors in your Rails application'
+description: 'Beware the remorseless robot army.'
+layout: article
+category: ruby
+image:
+  base: '2018/stop-robots-crawlers-triggering-errors-rails'
+  alt: 'BB-8'
+  source: 'https://unsplash.com/photos/C8VWyZhcIIU'
+  credit: 'Joseph Chan'
+---
+
+When a Rails application can't find a record, it throws a 404 error. This is a standard HTTP code for browsers (and other clients) meaning 'not found'.
+
+When you have a Internet-facing site various search engines will be crawling your site. As you change things certain URLs might no longer work and the search engines can start generating a lot of errors by hitting all the links that used to exist.
+
+
+## Instead of…
+
+Getting a bunch of unhelpful, distracting noise when Google (or another web crawler) hits deleted public pages…
+
+
+### Or
+
+...naively swallowing all your 404 errors.
+
+```ruby
+class ApplicationController < ActionController::Base
+  rescue_from ActiveRecord::RecordNotFound, with: :not_found
+
+  private
+
+  def not_found(exception)
+    render file:   Rails.root.join('public', '404.html'),
+           layout: nil,
+           status: :not_found
+  end
+end
+```
+
+
+## Use…
+
+A combination of the [`is_crawler`](https://github.com/ccashwell/is_crawler) gem and logic in your `application_controller.rb`.
```

Comment:
Choose one way: are you doing the "... <lowercase letter>" or "Capital"..when following on from the headings?

Do you know what I mean?

Reply:
Maybe: Use..the 'is_crawler' gem and required surrounding logic in your 'application_controller.rb'...

Line 44 is reading slightly confused...

Or maybe just: use the 'is_crawler' gem? After all, the example shows that you put the configuration in the application.rb?